src/Service/Locker.php

```diff
@@ -2,69 +2,67 @@
 
 namespace App\Service;
 
+use Doctrine\DBAL\Connection;
 use Doctrine\Persistence\ManagerRegistry;
 use App\Util\DoctrineTrait;
 
 class Locker
 {
     use DoctrineTrait;
 
-    private ManagerRegistry $doctrine;
-
-    public function __construct(ManagerRegistry $doctrine)
+    public function __construct(private ManagerRegistry $doctrine)
     {
-        $this->doctrine = $doctrine;
     }
 
-    public function lockPackageUpdate(int $packageId, int $timeout = 0)
+    public function lockPackageUpdate(int $packageId, int $timeout = 0): bool
     {
-        $this->getConn()->connect('master');
+        $this->getConn()->connect();
 
         return (bool) $this->getConn()->fetchOne('SELECT GET_LOCK(:id, :timeout)', ['id' => 'package_update_'.$packageId, 'timeout' => $timeout]);
     }
 
-    public function unlockPackageUpdate(int $packageId)
+    public function unlockPackageUpdate(int $packageId): void
     {
-        $this->getConn()->connect('master');
+        $this->getConn()->connect();
 
         $this->getConn()->fetchOne('SELECT RELEASE_LOCK(:id)', ['id' => 'package_update_'.$packageId]);
     }
 
-    public function lockSecurityAdvisory(string $source, int $timeout = 0)
+    public function lockSecurityAdvisory(string $source, int $timeout = 0): bool
     {
-        $this->getConn()->connect('master');
+        $this->getConn()->connect();
 
         return (bool) $this->getConn()->fetchOne('SELECT GET_LOCK(:id, :timeout)', ['id' => 'security_advisory_'.$source, 'timeout' => $timeout]);
     }
 
-    public function unlockSecurityAdvisory(string $source)
+    public function unlockSecurityAdvisory(string $source): void
     {
-        $this->getConn()->connect('master');
+        $this->getConn()->connect();
 
         $this->getConn()->fetchOne('SELECT RELEASE_LOCK(:id)', ['id' => 'security_advisory_'.$source]);
     }
 
-    public function lockCommand(string $command, int $timeout = 0)
+    public function lockCommand(string $command, int $timeout = 0): bool
     {
-        $this->getConn()->connect('master');
+        $this->getConn()->connect();
 
         return (bool) $this->getConn()->fetchOne(
             'SELECT GET_LOCK(:id, :timeout)',
             ['id' => $command, 'timeout' => $timeout]
         );
     }
 
-    public function unlockCommand(string $command)
+    public function unlockCommand(string $command): void
     {
-        $this->getConn()->connect('master');
+        $this->getConn()->connect();
 
         $this->getConn()->fetchOne(
             'SELECT RELEASE_LOCK(:id)',
             ['id' => $command]
         );
     }
 
-    private function getConn()
+    private function getConn(): Connection
     {
         return $this->getEM()->getConnection();
```

Comment:
The code above was written with the intent that this would return a PrimaryReadReplicaConnection potentially, and hence the `'master'` arg to force-connect the primary instance. IMO this should be left as is for safety, because while it is currently not using a replica setup, if we were to do it at some point this code would start breaking in weird ways. So it'd be best IMO to type the return here as PrimaryReadReplicaConnection and leave the rest untouched (except you can change master to primary now)

Reply:
Currently this returns `Doctrine\DBAL\Connection` and would break with runtime error
![Bildschirmfoto von 2022-01-23 13-54-21](https://user-images.githubusercontent.com/2852185/150679364-6b26b5b2-4b0d-4cb9-8994-93842f268a72.png)

keeping the `Connection` type would than raise:
```
Method Doctrine\DBAL\Connection::connect() invoked with 1 parameter, 0 required.
```

would you prefer to keep this warning in favor of switching to replica setup later?

Reply:
Hmm right :/ I guess the best way is to wrap all the calls above in a `if ($conn instanceof PrimaryReadReplicaConnection) { connect to  primary }` then?

Reply:
Because if the connection is a regular one, calling connect is completely useless here :)